File: utils/get_crosstable.R
library(argparse)
library(stringr)
#library(Seurat)
library(ArchR)
library(Matrix)
library(ggplot2)

#############################
parser <- ArgumentParser(description='Process some tasks')
parser$add_argument("--project",
                    type="character",
		    default=NULL,
                    help="the path of project saved")


parser$add_argument("--column",
                    type="character",
                    default="Clusters",
                    help="the column in metadata for cross table")


parser$add_argument("--outdir",
                    type="character",
                    default="ArchR_result")


parser$add_argument("--num_threads",
                    type="integer",
                    default=16,
                    help="number of threads for command")

args <- parser$parse_args()

options(stringsAsFactors=FALSE)
if(!dir.exists(args$outdir)){
        dir.create(args$outdir,recursive=TRUE)
}



print(paste0("Setting threads  :",args$num_threads))
addArchRThreads(threads = args$num_threads) 

print("# Loading ArrowFiles")
projHeme<-loadArchRProject(args$project)

meta=as.data.frame(getCellColData(projHeme))
if(!args$column%in%colnames(meta)){
	stop("the input column not in project's meta,please run update_metadata first!!!")
}

stopifnot("Sample"%in%colnames(meta))
m=table(meta[["Sample"]],meta[[args$column]])
write.table(as.matrix(m),file.path(args$outdir,paste0("Sample_",args$column,"_number.csv")),sep=",",quote=F)

stopifnot("Status"%in%colnames(meta))
m=table(meta[["Status"]],meta[[args$column]])
write.table(as.matrix(m),file.path(args$outdir,paste0("Status_",args$column,"_number.csv")),sep=",",quote=F)

File: utils/get_crosstable.py
import argparse
import os
import scanpy as sc
import pandas as pd


parser=argparse.ArgumentParser()
parser.add_argument("--outdir",type=str,default=None)
parser.add_argument("--data",type=str,default=None)
parser.add_argument("--column",type=str,default="leiden")



args=parser.parse_args()
if not os.path.exists(args.outdir):
   os.makedirs(args.outdir)

print("*** Loading data")
adata=sc.read_h5ad(args.data)
metadata=adata.obs

if args.column in metadata.columns:
    if "Sample" in metadata.columns:
        df=pd.crosstab(metadata["Sample"],metadata[args.column])
        df.to_csv(os.path.join(args.outdir,"sample_"+args.column+"_crosstable.csv"),sep=",")

    if "Status" in metadata.columns:
        df=pd.crosstab(metadata["Status"],metadata[args.column])
        df.to_csv(os.path.join(args.outdir,"status_"+args.column+"_crosstable.csv"),sep=",")

else:
    raise ValueError("Please input valid column in metadata")
